src/main/java/org/prebid/server/vertx/http/CircuitBreakerSecuredHttpClient.java

```diff
@@ -68,8 +73,37 @@ private void circuitClosed(String name) {
         final String name = nameFrom(url);
         final CircuitBreaker breaker = circuitBreakerByName.computeIfAbsent(name, circuitBreakerCreator);
 
-        return breaker.<HttpClientResponse>execute(future -> httpClient.request(method, url, headers, body, timeoutMs)
-                .setHandler(future));
+        return breaker.execute(future -> httpClient.request(method, url, headers, body, timeoutMs)
+                .compose(response -> succeedBreaker(response, future))
+                .recover(exception -> failBreaker(exception, future, name, breaker)));
+    }
+
+    private static Future<HttpClientResponse> succeedBreaker(HttpClientResponse response,
+                                                             Future<HttpClientResponse> future) {
+        future.complete(response);
+        return future;
+    }
+
+    private Future<HttpClientResponse> failBreaker(Throwable exception, Future<HttpClientResponse> future,
+                                                   String name, CircuitBreaker breaker) {
+        ensureToIncrementFailureCount(name, breaker);
+
+        future.fail(exception);
+        return future;
+    }
+
+    /**
+     * Reset failure counter to adjust open-circuit time frame.
+     */
+    private void ensureToIncrementFailureCount(String name, CircuitBreaker breaker) {
+        final long currentTimeMillis = System.currentTimeMillis();
+
+        final long lastFailureMs = lastFailureByName.computeIfAbsent(name, ignored -> currentTimeMillis);
```

Comment:
How do you think, is it possible to put reset logic into a wrapper over Verx CircuitBreaker?

Reply:
Yes, it is possible. But we don't have the case when need to keep continuously increasing failure counter so far.